wrappers/golang_v3/curves/bn254/curve.go

```diff
@@ -0,0 +1,171 @@
+	cP := (*C.projective_t)(unsafe.Pointer(&p))
+	C.bn254_to_affine(cP, cA)
+	return a
+}
+
+func GenerateProjectivePoints(size int) core.HostSlice[Projective] {
+	points := make([]Projective, size)
+	for i := range points {
+		points[i] = Projective{}
+	}
+
+	pointsSlice := core.HostSliceFromElements[Projective](points)
+	pPoints := (*C.projective_t)(unsafe.Pointer(&pointsSlice[0]))
+	cSize := (C.int)(size)
+	C.bn254_generate_projective_points(pPoints, cSize)
+
+	return pointsSlice
+}
+
+type Affine struct {
+	X, Y BaseField
+}
+
+func (a Affine) Size() int {
+	return a.X.Size() * 2
+}
+
+func (a Affine) AsPointer() *uint32 {
+	return a.X.AsPointer()
+}
+
+func (a *Affine) Zero() Affine {
+	a.X.Zero()
+	a.Y.Zero()
+
+	return *a
+}
+
+func (a *Affine) FromLimbs(x, y []uint32) Affine {
+	a.X.FromLimbs(x)
+	a.Y.FromLimbs(y)
+
+	return *a
+}
+
+func (a Affine) ToProjective() Projective {
+	var z BaseField
+
+	return Projective{
+		X: a.X,
+		Y: a.Y,
+		Z: z.One(),
+	}
+}
+
+func AffineFromProjective(p *Projective) Affine {
+	return p.ProjectiveToAffine()
+}
+
+func GenerateAffinePoints(size int) core.HostSlice[Affine] {
+	points := make([]Affine, size)
+	for i := range points {
+		points[i] = Affine{}
+	}
+
+	pointsSlice := core.HostSliceFromElements[Affine](points)
+	cPoints := (*C.affine_t)(unsafe.Pointer(&pointsSlice[0]))
+	cSize := (C.int)(size)
+	C.bn254_generate_affine_points(cPoints, cSize)
+
+	return pointsSlice
+}
+
+func convertAffinePointsMontgomery(points *core.DeviceSlice, isInto bool) runtime.EIcicleError {
+	defaultCfg := core.DefaultVecOpsConfig()
+	cValues, _, _, cCfg, cSize := core.VecOpCheck(*points, *points, *points, &defaultCfg)
+	cErr := C.bn254_affine_convert_montgomery((*C.affine_t)(cValues), (C.size_t)(cSize), (C._Bool)(isInto), (*C.VecOpsConfig)(cCfg), (*C.affine_t)(cValues))
+	err := runtime.EIcicleError(cErr)
+	return err
+}
+
+func AffineToMontgomery(points *core.DeviceSlice) runtime.EIcicleError {
+	points.CheckDevice()
+	return convertAffinePointsMontgomery(points, true)
+}
+
+func AffineFromMontgomery(points *core.DeviceSlice) runtime.EIcicleError {
+	points.CheckDevice()
+	return convertAffinePointsMontgomery(points, false)
+}
+
+func convertProjectivePointsMontgomery(points *core.DeviceSlice, isInto bool) runtime.EIcicleError {
+	defaultCfg := core.DefaultVecOpsConfig()
+	cValues, _, _, cCfg, cSize := core.VecOpCheck(*points, *points, *points, &defaultCfg)
+	cErr := C.bn254_projective_convert_montgomery((*C.projective_t)(cValues), (C.size_t)(cSize), (C._Bool)(isInto), (*C.VecOpsConfig)(cCfg), (*C.projective_t)(cValues))
+	err := runtime.EIcicleError(cErr)
+	return err
+}
+
+func ProjectiveToMontgomery(points *core.DeviceSlice) runtime.EIcicleError {
+	points.CheckDevice()
+	return convertProjectivePointsMontgomery(points, true)
+}
+
+func ProjectiveFromMontgomery(points *core.DeviceSlice) runtime.EIcicleError {
+	points.CheckDevice()
+	return convertProjectivePointsMontgomery(points, false)
```

Comment:
The backends support on host as well now so we can remove the restriction of being DeviceSlice